### PR TITLE
[ADD] purchase_order_product_recommendation_brand: New module

### PR DESCRIPTION
**Pull request by chienandalu**
Allows to filter product recommendations by brand.

Depends on 

- [x] https://github.com/OCA/purchase-workflow/pull/805 purchase_order_product_recommendation domain  method

cc @Tecnativa TT20723

### CODE REVIEW

File: purchase_order_product_recommendation_brand/wizards/purchase_order_recommendation.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        comodel_name='product.brand',
+        string="Brands",
+    )
+
+    def _get_products(self):
+        """Filter products of the given brands"""
+        products = super()._get_products()
+        # Filter products by brand if set.
+        # It will apply to show_all_partner_products as well
+        if self.product_brand_ids:
```
Comment (pedrobaeza) [inline purchase_order_product_recommendation_brand/wizards/purchase_order_recommendation.py:19]: Why do you need this if you are already filtering in the domain?

Comment (chienandalu) [reply purchase_order_product_recommendation_brand/wizards/purchase_order_recommendation.py:19]: In this method , I filter the supplier products (given by the proper supplierinfos). In `_get_all_products_domain` we obtain all the products independently of the supplier.

Comment (pedrobaeza) [reply purchase_order_product_recommendation_brand/wizards/purchase_order_recommendation.py:19]: OK, I'm not sure about the performance of this.

**Review by pedrobaeza — APPROVED** [review]

**Review by rafaelbn — APPROVED** [review]

**Comment by pedrobaeza** [comment]
I'm thinking this module should go to OCA/brand on 12.0

**Comment by chienandalu** [comment]
> I'm thinking this module should go to OCA/brand on 12.0

I didn't know the repo, but anyway `product-attribute/product_brand` and `brand/brand` doesn't seem to have any relation, don't they?

**Comment by chienandalu** [comment]
ok, now I see product_brand in the repo, but in the v13 branch

**Comment by pedrobaeza** [comment]
Yes, it was moved later. We can wait for v13 if you think so.

**Pull request merged**